roles/common/tasks/configure_logredactor.yml

```diff
@@ -0,0 +1,44 @@
+---
+- name: Get the logger definition to check if redactor added
+  ansible.builtin.shell: grep {{ item.logger_name }} {{ log4j_file }}
+  register: output
+  failed_when: output.rc == 2
+
+- name: Add redactor in the logger if not already added
+  ansible.builtin.replace:
+    path: "{{log4j_file}}"
+    regexp: '{{ item.logger_name }}=(.*)'
+    replace: '{{ item.logger_name }}=\1, {{ item.redactor_name }}'
+  when: not ( output | regex_search("{{ item.redactor_name }}") )
```

Comment:
To solve this and make it deterministic, 
In the task above this one, 
We'll now `grep ^{{ item.logger_name }}= {{ log4j_file }}` instead of `grep {{ item.logger_name }} {{ log4j_file }}`
This'll ensure only the logger definition is grepped, if logger is available. Since logger can be defined only once in the file. And if a line starts with that and not #, it certainly is the logger definition. 

Now in this task, the regex search would definitely work. Since, we'll search for the redactor name anywhere in the logger definition (it can be anywhere after the 'logger_name=')